scripts/src/java/org/oppia/android/scripts/ci/ComputeAffectedTests.kt

```diff
@@ -103,10 +103,18 @@ private fun computeAffectedTargetsForNonDevelopBranch(
   println("Affected test targets due to transitive build deps: $transitiveTestTargets")
 
   val allAffectedTestTargets = (affectedTestTargets + transitiveTestTargets).toSet()
+
+  // Filtering out the targets to be ignored.
+  val filteredAffectedTestTargets = allAffectedTestTargets.filter { targetPath ->
+    !targetPath.startsWith("//instrumentation", ignoreCase = true)
+  }
```

Comment:
Please add a test to cover this behavioral change (to make sure that instrumentation test targets are correctly ignored).

Reply:
I've added a test regarding this I'm not sure why it fails.
It works in CI as expected.

aren't we using the ComputedAffectedTests class/script in the test?

Reply:
Yes, it's using the same routine. This may be due to some misconfiguration in the test.

Edit: I think this failing test is catching a real scenario: the ComputeAffectedTests script will run *all* tests when on the develop branch (your test never changes the branch). To remediate, I suggest:
1. Introducing a new test that changes to a feature branch but otherwise creates the same setup (it should pass)
2. Fix the ``develop`` test case by adding the missing logic to the script since it would otherwise fail after merging the PR

I also highly suggest running these tests locally & not waiting for CI as that will significantly slow you down.

Reply:
Done, Thanks